BrowseRouter/ConfigService.cs

```diff
@@ -8,18 +8,32 @@ public interface IConfigService
 public class ConfigService : IConfigService
 {
   /// <summary>
-  /// Config lives in the same folder as the EXE, name "BrowserSelector.ini".
+  /// Config lives in the same folder as the EXE, name "config.ini".
   /// </summary>
-  public readonly string ConfigPath;
+  private string? configPath = null;
+  private string ConfigPath => configPath ??= Path.Combine(Path.GetDirectoryName(App.ExePath)!, "config.ini");
 
-  public ConfigService()
+  private IEnumerable<string>? configLines = null;
+  private IEnumerable<string> ConfigLines
   {
-    // Fix for self-contained publishing
-    this.ConfigPath = Path.Combine(Path.GetDirectoryName(App.ExePath)!, "config.ini");
+    get
+    {
+      if (configLines != null)
+        return configLines;
+
+      if (!File.Exists(ConfigPath))
```

Comment:
I appreciate the added caching. Big getters are not good style. Could you please move the majority of this getter back into `ReadFile` ?

Reply:
Strictly speaking, this is an added feature and belongs in a separate PR.